Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/CompeteController.cs

```diff
@@ -452,11 +452,16 @@ public ActionResult Submit(SubmissionModel participantSubmission, bool official)
             {
                 throw new HttpException((int)HttpStatusCode.BadRequest, Resource.ContestsGeneral.Submission_too_long);
             }
-
+            
             if (this.Data.Submissions.HasUserNotProcessedSubmissionForProblem(problem.Id, this.UserProfile.Id))
             {
                 throw new HttpException((int)HttpStatusCode.BadRequest, Resource.ContestsGeneral.User_has_not_processed_submission_for_problem);
             }
+            
+            if (participant.Contest.UsersCantSubmitConcurrently && this.Data.Submissions.UserHasUnprocessedSubmissionInContest(participant.ContestId, this.UserProfile.Id))
+            {
+                return this.JsonError(Resource.ContestsGeneral.User_has_not_processed_submission_for_contest);
```

Comment:
Currently any error is displayed as just "error" on the client because we return BusinessException which cannot be parsed by the client properly. Theres a requirement in the issue for a specific message, so thus why im using JsonError. You can see the submission-page.js file for more info.